api/src/helpers.py

```diff
@@ -15,82 +16,89 @@
 
 
 def trigger_discover(input: Dict) -> Dict:
-    MWAA_ENV = os.environ["MWAA_ENV"]
-    airflow_client = boto3.client("mwaa")
-    mwaa_cli_token = airflow_client.create_cli_token(Name=MWAA_ENV)
+    if MWAA_ENV := os.environ["MWAA_ENV"]:
+        airflow_client = boto3.client("mwaa")
+        mwaa_cli_token = airflow_client.create_cli_token(Name=MWAA_ENV)
 
-    mwaa_webserver_hostname = (
-        f"https://{mwaa_cli_token['WebServerHostname']}/aws_mwaa/cli"
-    )
+        mwaa_webserver_hostname = (
+            f"https://{mwaa_cli_token['WebServerHostname']}/aws_mwaa/cli"
+        )
 
-    unique_key = str(uuid4())
-    raw_data = f"dags trigger veda_discover --conf '{input.json()}' -r {unique_key}"
-    mwaa_response = requests.post(
-        mwaa_webserver_hostname,
-        headers={
-            "Authorization": "Bearer " + mwaa_cli_token["CliToken"],
-            "Content-Type": "application/json",
-        },
-        data=raw_data,
-    )
-    if mwaa_response.status_code not in [200, 201]:
-        raise Exception(
-            f"Failed to trigger airflow: {mwaa_response.status_code} {mwaa_response.text}"
+        unique_key = str(uuid4())
+        raw_data = f"dags trigger veda_discover --conf '{input.json()}' -r {unique_key}"
+        mwaa_response = requests.post(
+            mwaa_webserver_hostname,
+            headers={
+                "Authorization": "Bearer " + mwaa_cli_token["CliToken"],
+                "Content-Type": "application/json",
+            },
+            data=raw_data,
         )
+        if mwaa_response.status_code not in [200, 201]:
```

Comment:
Is this more semantic?

```suggestion
        if not mwaa_response.ok:
```

Reply:
I think there's also a response.raise() method.